### DISCUSSION TITLE
Question about "Passing arguments and this to listeners" section

### DISCUSSION THREAD

**Discussion by iggeehu** [nodejs/help#5035] [Q&A] [answered]
### Affected URL(s)

https://nodejs.org/api/events.html#class-eventemitter

### Description of the problem

Hi guys, why does "this" in the listener function log as the class MyEmitter, rather than the instance myEmitter, of the class?
The text said that the "this" keyword is set to the EventEmitter instance to which the listener is attached, but isn't the listener attached to myEmitter instead?

This is the text on the doc
"
The eventEmitter.emit() method allows an arbitrary set of arguments to be passed to the listener functions. Keep in mind that when an ordinary listener function is called, the standard this keyword is intentionally set to reference the EventEmitter instance to which the listener is attached.

```
const myEmitter = new MyEmitter();
myEmitter.on('event', function(a, b) {
  console.log(a, b, this, this === myEmitter);
  // Prints:
  //   a b MyEmitter {
  //     _events: [Object: null prototype] { event: [Function (anonymous)] },
  //     _eventsCount: 1,
  //     _maxListeners: undefined,
  //     [Symbol(kCapture)]: false
  //   } true
});
myEmitter.emit('event', 'a', 'b');
```

Thanks!

**Comment by mscdex** [comment, marked as answer]
`console.log()` shows the class name where available to let you know the type of object you're logging. In this case, `myEmitter` is an instance of `MyEmitter`, so it shows that class name. The properties shown in the output are those that belong to that specific instance (`this`) and not the class/constructor.
